code/parachain/frame/currency-factory/src/lib.rs

```diff
@@ -210,9 +210,20 @@ pub mod pallet {
 		}
 	}
 
+	impl<T: Config> composable_traits::currency::AssetDataMutate for Pallet<T> {
+		type AssetId = T::AssetId;
+		type Balance = T::Balance;
+
+		fn update_existential_deposit(asset_id: Self::AssetId, ed: Option<Self::Balance>) {
+			AssetEd::<T>::set(asset_id, ed)
+		}
+	}
+
 	impl<T: Config> LocalAssets<T::AssetId> for Pallet<T> {
+		// NOTE: it is not true as of now, so we should not rely on chain for this.
+		// NOTE: XCM does not supports decimals and Statemine can have assets without decimals
+		// NOTE: hence we cannot default to 12 decimals robustly 
```

Comment:
What do you recommend as an alternative?

Reply:
remove all 12 decimals here and fix all places - not in this PR.